activation/poetdb.go

```diff
@@ -83,31 +86,45 @@ func (db *PoetDb) storeProof(proofMessage *types.PoetProofMessage) error {
 		return fmt.Errorf("failed to get poet proof message reference: %v", err)
 	}
 
+	if err := db.storeProofInSQLDB(proofMessage, ref); err != nil {
+		return err
+	}
+
+	db.log.With().Info("stored poet proof",
+		log.String("poet_proof_id", fmt.Sprintf("%x", ref[:5])),
+		log.String("round_id", proofMessage.RoundID),
+		log.String("poet_service_id", fmt.Sprintf("%x", proofMessage.PoetServiceID[:5])),
+	)
+
+	key := makeKey(proofMessage.PoetServiceID, proofMessage.RoundID)
+	db.publishProofRef(key, ref)
+
+	return nil
+}
+
+func (db *PoetDb) storeProofInSQLDB(proofMessage *types.PoetProofMessage, ref []byte) error {
 	messageBytes, err := types.InterfaceToBytes(proofMessage)
 	if err != nil {
 		return fmt.Errorf("could not marshal proof message: %v", err)
 	}
 
-	batch := db.store.NewBatch()
-	if err := batch.Put(ref, messageBytes); err != nil {
-		return fmt.Errorf("failed to store poet proof for poetId %x round %s: %v",
-			proofMessage.PoetServiceID[:5], proofMessage.RoundID, err)
+	dbtx, err := db.sqlDB.Tx(context.Background())
+	if err != nil {
+		return err
 	}
-	key := makeKey(proofMessage.PoetServiceID, proofMessage.RoundID)
-	if err := batch.Put(key[:], ref); err != nil {
-		return fmt.Errorf("failed to store poet proof index entry for poetId %x round %s: %v",
+
+	defer dbtx.Release()
+
+	if err := poets.Add(dbtx, ref, messageBytes, proofMessage.PoetServiceID, proofMessage.RoundID); err != nil {
```

Comment:
why does this need a dbtx if there is only one stmt to execute?

Reply:
Forgot to remove it after resolving a CR request. Thanks. Removed